cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -58,37 +58,49 @@ readonly USE_CLOUD_SQL_PRIVATE_IP
 METASTORE_INSTANCE="$(/usr/share/google/get_metadata_value attributes/hive-metastore-instance || echo '')"
 readonly METASTORE_INSTANCE
 
-# Get metastore DB instance type, result be one of MYSQL, POSTGRES,
+ADDITIONAL_INSTANCES="$(/usr/share/google/get_metadata_value ${ADDITIONAL_INSTANCES_KEY} || echo '')"
+readonly ADDITIONAL_INSTANCES
+
+# Get metastore DB instance type, result be one of MYSQL, POSTGRES, SQLSERVER
 function get_metastore_instance_type() {
-  local metastore_instance="${METASTORE_INSTANCE}"
+  local instances=$1
+  local metastore_instance=$(echo ${instances} | cut -d "," -f 1)
+  local database=''
   if [[ -z "${metastore_instance}" ]]; then
-    err 'Must specify hive-metastore-instance VM metadata'
-  fi
-  if ! [[ "${metastore_instance}" =~ .+:.+:.+ ]]; then
-    err 'hive-metastore-instance must be of form project:region:instance'
-  fi
-  local project=${metastore_instance%*:*:*}
-  local instance=${metastore_instance##*:}
-  local database=$(gcloud sql instances describe --project=${project} ${instance} | grep 'databaseVersion')
-  if [[ -z "${database}" ]]; then
-    err 'Unable to find hive_metastore_instance'
+    err 'Must specify metastore-instance VM metadata'
+  elif ! [[ "${metastore_instance}" =~ .+:.+:.+ ]]; then
+    err 'metastore-instance must be of form project:region:instance'
+  else
+    local project=${metastore_instance%*:*:*}
+    local instance=${metastore_instance##*:}
+    database=$(gcloud sql instances describe --project=${project} ${instance} | grep 'databaseVersion')
+    if [[ -z "${database}" ]]; then
+      err 'Unable to find metastore_instance'
+    else
+      # Trim off version and whitespaces and use upper case
+      # databaseVersion: MYSQL_8_0
+      # databaseVersion: POSTGRES_12
+      # databaseVersion: SQLSERVER_2019_STANDARD
+      database=${database##*:}
+      database=${database%%_*}
+      database="${database#"${database%%[![:space:]]*}"}"
+    fi
   fi
-  # Trim off version and whitespaces and use upper case
-  # databaseVersion: MYSQL_8_0
-  # databaseVersion: POSTGRES_12
-  # databaseVersion: SQLSERVER_2019_STANDARD
-  database=${database##*:}
-  database=${database%%_*}
-  database="${database#"${database%%[![:space:]]*}"}"
   echo "${database^^}"
 }
 
-# CLOUD SQL instance type is one of MYSQL, POSTGRES, SQL
-METASTORE_INSTANCE_TYPE=$(get_metastore_instance_type)
+# CLOUD SQL instance type is one of MYSQL, POSTGRES, SQLSERVER.
+# Type inferred in order of METASTORE_INSTANCE, ADDITIONAL_INSTANCES, default to MYSQL
+if [[ -n "${METASTORE_INSTANCE}" ]]; then
+  METASTORE_INSTANCE_TYPE=$(get_metastore_instance_type "${METASTORE_INSTANCE}")
+elif [[ -n "${ADDITIONAL_INSTANCES}" ]]; then
+  METASTORE_INSTANCE_TYPE=$(get_metastore_instance_type "${ADDITIONAL_INSTANCES}")
+fi
```

Comment:
As discussed offline, we should not get type from additional instances, instead add a metadata for type, if not provided, default to MYSQL.

Reply:
added metadata cloud-sql-instance-type that takes priority